File: ordered-receiver/src/main/java/com/example/broadcast_types/MainActivity.java
package com.example.broadcast_types;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import com.example.broadcast_types.databinding.ActivityMainBinding;

/**
 * receiver action names must be same,
 * need to use sendOrderedBroadcast() instead of sendBroadcast()
 */
public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());


        binding.buttonRun.setOnClickListener(view ->{
            sendOrderedBroadcast(new Intent(this, ReceiverB.class), null);
            sendOrderedBroadcast(new Intent(this, ReceiverC.class), null);
            sendOrderedBroadcast(new Intent(this, ReceiverA.class), null);
        });
    }
}

File: local-receiver/src/main/java/com/fcenesiz/local_receiver/MainActivity.kt
package com.fcenesiz.local_receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.fcenesiz.local_receiver.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding : ActivityMainBinding
    lateinit var localBroadcastManager: LocalBroadcastManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        localBroadcastManager = LocalBroadcastManager.getInstance(this)
        binding.buttonSendBroadcast.setOnClickListener {
            sendBroadcast()
        }
    }

    private fun sendBroadcast(){
        val intent = Intent(this, MyReceiver::class.java)
        intent.putExtra("x", 15)
        intent.putExtra("y", 20)
        sendBroadcast(intent)
    }

    private val returnReceiver = object : BroadcastReceiver(){
        override fun onReceive(context: Context?, intent: Intent?) {
            val sum = intent!!.getIntExtra("sum", 0)
            Toast.makeText(baseContext, "sum: $sum", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onResume() {
        super.onResume()
        val intentFilter = IntentFilter("my.result.receiver.action")
        localBroadcastManager.registerReceiver(returnReceiver, intentFilter)
    }

    override fun onPause() {
        super.onPause()
        unregisterReceiver(returnReceiver)
    }

}